### PR TITLE
Added release criteria to release-instructions

### PR DESCRIPTION
**Pull request by dsu-igeek**
Signed-off-by: Dave Smith-Uchida <dsmithuchida@vmware.com>

Thank you for contributing to Velero!

# Please add a summary of your change

# Does your change fix a particular issue?

Fixes #(issue)

# Please indicate you've done the following:

- [ ] [Accepted the DCO](https://velero.io/docs/v1.5/code-standards/#dco-sign-off). Commits without the DCO will delay acceptance.
- [ ] [Created a changelog file](https://velero.io/docs/v1.5/code-standards/#adding-a-changelog) or added `/kind changelog-not-required`.
- [ ] Updated the corresponding documentation in `site/content/docs/main`.

### CODE REVIEW

**Review by ashish-amarnath — APPROVED** [review]

File: site/content/docs/main/release-instructions.md [hunk 1]
```diff
@@ -14,6 +14,32 @@ This page covers the steps to perform when releasing a new version of Velero.
 - Pre-releases: Any release leading up to a GA. Example: 1.4.0-beta.1, 1.5.0-rc.1
 - RC releases: Release Candidate, contains everything that is supposed to ship with the GA release. This is still a pre-release.
 
+# Velero Release Requirements
```
Comment (carlisia) [inline site/content/docs/main/release-instructions.md:17]: This header needs to be done with "##" and the headers below this one need to be done with "###". the TOC is not picking up this header.

File: site/content/docs/main/release-instructions.md [hunk 2]
```diff
@@ -14,6 +14,32 @@ This page covers the steps to perform when releasing a new version of Velero.
...
+Velero is on a "train leaves the station" model for releases.  We will generate a release candidate (RC)
+at the scheduled time.  Multiple release candidates may be generated, depending on if bugs are found during testing.
+When testing has passed a release build will be generated.
+
+## Release Candidate criteria
+The release candidate commit must meet the following criteria:
+
+* No major bugs outstanding 
+* Unit tests pass
+* E2E tests against latest K8S on AWS, vSphere pass
```
Comment (carlisia) [inline site/content/docs/main/release-instructions.md:28]: Super nit but K8S is not an acronym, so it's weird to see it capitalized. Either s/K8S/K8s or s/K8S/k8s, thank you.

**Review by carlisia — CHANGES_REQUESTED** [review]
I was reading this to do a checklist and noticed a couple things.

File: site/content/docs/main/release-instructions.md (hunk 2 above)
Comment (carlisia) [reply site/content/docs/main/release-instructions.md:28]: Also, shouldn't Kind be on this list?

File: site/content/docs/main/release-instructions.md (hunk 1 above)
Comment (dsu-igeek) [reply site/content/docs/main/release-instructions.md:17]: Thanks

File: site/content/docs/main/release-instructions.md (hunk 2 above)
Comment (dsu-igeek) [reply site/content/docs/main/release-instructions.md:28]: Sure, Kind is a good idea.  They seem to have it as "kind" so that's what I'm putting in here.

**Review by jenting — APPROVED** [review]

**Pull request merged**